src/overlays/actors/ovl_En_Kakasi2/z_en_kakasi2.c

```diff
@@ -20,21 +42,193 @@ const ActorInit En_Kakasi2_InitVars = {
     (ActorFunc)EnKakasi2_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/EnKakasi2_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi2/EnKakasi2_Destroy.s")
+void EnKakasi2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi2* this = THIS;
+    s32 pad;
+    f32 argument0;
+    f32 zAngle;
+
+    osSyncPrintf("\n\n");
+    // Visit Umeda
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 梅田参号見参！ ☆☆☆☆☆ \n" VT_RST);
+
+    this->switchFlag = this->actor.params & 0x3F;
+    argument0 = (this->actor.params >> 6) & 0xFF;
+    zAngle = this->actor.posRot.rot.z;
```

Comment:
confusingly, they often use rotation as an extension of params when there isnt enough room. so in this case its not actually a rotation value. I would rename the temp to something more generic until we understand what the different params are to name it proper

Reply:
👍